src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -760,24 +760,34 @@ private async Task ReplaceIndicatorFootnote(Guid footnoteId, TargetableReplaceme
             });
         }
 
-        private async Task<Either<ActionResult, Unit>> RemoveOriginalSubjectAndFileFromRelease(Guid releaseId,
-            Guid originalFileId, Guid replacementFileId)
+        private async Task<Either<ActionResult, Unit>> RemoveOriginalSubjectAndFileFromRelease(
+            Guid releaseId,
+            Guid originalFileId,
+            Guid replacementFileId)
         {
             // First, unlink the original file from the replacement before removing it.
             // Ordinarily, removing a file from a Release deletes any associated replacement
             // so that there's no possibility of abandoned replacements being orphaned from their original files.
             return await CheckReleaseFileReferenceExists(originalFileId)
                 .OnSuccessVoid(async originalFile =>
                 {
-                    if (originalFile.ReplacedById != replacementFileId)
-                    {
-                        throw new InvalidOperationException(
-                            $"Expected the original file reference to be associated with the replacement but found: {originalFile.ReplacedById}");
-                    }
-
-                    _contentDbContext.Update(originalFile);
-                    originalFile.ReplacedById = null;
-                    await _contentDbContext.SaveChangesAsync();
+                    await CheckReleaseFileReferenceExists(replacementFileId)
+                        .OnSuccessVoid(
+                            async replacementFile =>
+                            {
+                                if (originalFile.ReplacedById != replacementFile.Id)
+                                {
+                                    throw new InvalidOperationException(
+                                        $"Expected the original file reference to be associated with the replacement but found: {originalFile.ReplacedById}");
+                                }
+
+                                originalFile.ReplacedById = null;
+                                replacementFile.ReplacingId = null;
+                                _contentDbContext.Update(originalFile);
+                                _contentDbContext.Update(replacementFile);
```

Comment:
There was a little test around this which could be extended to include the replacement file

Look for this in ReplacementServiceTests
```
                // Check that the original file was unlinked from the replacement before the mock call to remove it.
                Assert.Null((await contentDbContext.ReleaseFileReferences.FindAsync(originalReleaseFileReference.Id))
                    .ReplacedBy);
```

Reply:
Added the additional assertion on the replacement file